pkg/api/object_example_test.go

```diff
@@ -39,7 +40,43 @@ type ExampleObject struct {
 // identifier is appended by default for Get, Update and Destroy operations. You can implement the interface
 // types.RequestFilterHook to have full control over the requests done for your object.
 func (o *ExampleObject) EndpointURL(ctx context.Context) (*url.URL, error) {
-	return url.Parse("/example/v1")
+	op, err := types.OperationFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	endpointSuffix := "foo.json"
+	if op == types.OperationGet && o.Identifier == "" && o.SecondaryIdentifier != "" {
+		endpointSuffix = "foo/by-secondary-identifier.json"
+	}
+
+	return url.Parse(fmt.Sprintf("/example/v1/%s", endpointSuffix))
+}
+
+// GetIdentifier returns the identifier of an object.
+// The Makefiles `generate` target creates this method for objects that have a anxcloud:"identifier" tagged field.
+// Objects with multiple identifying fields can have this method overwritten to return an operation dependent identifier.
+// The provided context might not always have the operation set. In this case the method must return the primary identifier.
+//
+// For example: `corev1.Location` has the (primary) `Identifier` attribute, which is required by many API endpoints.
+// Additionally locations are identified by the human readable `Code` attribute (e.g. ANX04).
+//
+// Handling of secondary identifiers usually requires additional logic in the objects EndpointURL method.
+func (o *ExampleObject) GetIdentifier(ctx context.Context) (string, error) {
+	if o.Identifier != "" {
+		return o.Identifier, nil
+	}
+
+	op, err := types.OperationFromContext(ctx)
+	if err != nil {
+		return "", nil
+	}
```

Comment:
first I stumbled over the `return "", nil` in error case, but then I remembered this is just the primary identifiers right now. We should make this more clear, but I think we can even improve the code some more:

```go
op, err := types.OperationFromContext(ctx)
if o.Identifier != "" || err != nil {
  return o.Identifier, nil
}

// [...]
```